src/compiler/scala/tools/nsc/classpath/DirectoryClassPath.scala

```
@@ -125,13 +126,31 @@ trait JFileDirectoryLookup[FileEntryType <: ClassRepresentation] extends Directo
 
 object JrtClassPath {
   import java.nio.file._, java.net.URI
-  def apply(): Option[ClassPath] = {
-    try {
-      val fs = FileSystems.getFileSystem(URI.create("jrt:/"))
-      Some(new JrtClassPath(fs))
-    } catch {
-      case _: ProviderNotFoundException | _: FileSystemNotFoundException =>
-        None
+  def apply(release: Option[String]): Option[ClassPath] = {
+    import scala.util.Properties._
+    def jrt = {
+      try {
+        val fs = FileSystems.getFileSystem(URI.create("jrt:/"))
+        Some(new JrtClassPath(fs))
+      } catch {
+        case _: ProviderNotFoundException | _: FileSystemNotFoundException =>
+          None
+      }
+    }
+    release match {
+      case None => jrt
+      case Some(v) if v == javaSpecVersion =>
+        if (isJavaAtLeast("9")) jrt else None
+      case Some(older) =>
+        if (isJavaAtLeast("9")) {
+          try {
+            Some(new CtSymClassPath(release))
+          } catch {
+            case _: Throwable =>
+              // TODO more fine grained error handling
+              None
```

we probably want to report an error if we cannot satisfy what the users asks for using the `-release` flag